examples/bsoncxx/builder_basic.cpp

```diff
@@ -0,0 +1,60 @@
+#include <bsoncxx/builder/basic/array.hpp>
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/builder/basic/kvp.hpp>
+#include <bsoncxx/types.hpp>
+
+using namespace bsoncxx;
+
+int main(int, char**) {
+
+    // bsoncxx::builder::basic presents a BSON-construction interface familiar to users of the server's
+    // BSON library or the Java driver.
+
+    // basic::document builds a BSON document.
+    auto doc = builder::basic::document{};
+    // basic::array builds a BSON array
+    auto arr = builder::basic::array{};
+
+    // We append key-value pairs to a document using the kvp helper.
+    using bsoncxx::builder::basic::kvp;
+
+    doc.append(kvp("foo", "bar")); // string literal value will be converted to b_utf8 automatically
+    doc.append(kvp("baz", types::b_bool{false}));
+    doc.append(kvp("garply", types::b_double{3.14159}));
+
+    // We can also pass a variable number of keys to append.
+    doc.append(kvp("a key", "a value"),
+               kvp("another key", "another value"),
+               kvp("moar keys", "moar values"));
+
+    // Appending to arrays is simple, just append one or more bson values.
+    arr.append("hello");
+    arr.append(false, types::b_bool{true}, types::b_double{1.234});
+
+    // If we want to create a subdocument, we can pass lambda as a value with a sub_document argument.
```

Comment:
Is this the only way? Maybe append arr to doc above to show that using a lambda is only an option?